sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -101,4 +102,30 @@ ng_netreg_entry_t *ng_netreg_getnext(ng_netreg_entry_t *entry)
     return entry;
 }
 
+int ng_netreg_calculate_csum(ng_nettype_t pseudo_hdr_type,
+                             ng_nettype_t csum_hdr_type,
+                             ng_pktsnip_t *pkt)
+{
+    switch (csum_hdr_type) {
+#ifdef MODULE_NG_ICMPV6
```

Comment:
The bigger problem is not that the functions are not defined, but the different values for `ng_nettype_t` for different modules used, due to size optimization for the `ng_netreg` look-up table. If the UDP module does not exist at compile time, `NG_NETTYPE_UDP` is not defined for example.